lib/rules/no-shadow.js

```diff
@@ -117,6 +117,29 @@ module.exports = {
             return def && def.name.range;
         }
 
+        /**
+         * Get declared line and column of a variable.
+         * @param {eslint-scope.Variable} variable The variable to get.
+         * @returns {Object|undefined} The declared line and column of the variable.
+         */
+        function getDeclaredLocation(variable) {
+            const identifier = variable.identifiers[0];
+            let obj;
+
+            if (identifier) {
+                obj = {
+                    line: identifier.loc.start.line,
+                    column: identifier.loc.start.column + 1
+                };
+            } else {
+                obj = {
+                    line: 0,
+                    column: 0
```

Comment:
For global variables that have no location within the file to point to, pointing to  `0:0` seems a bit strange. What do you think of adding a second message saying something like "___ is already a global variable." without location information, but the "upper scope" message does since we know where to point to?

Reply:
Thank you for your comment! I understood.
I'll change codes to show message that the variable is a global variable, instead of no location `0:0`